// datebase/restaurantDB.js
/*var mysql = require('mysql');

var pool = mysql.createPool({
  connectionLimit: 100,
  host: "localhost",
  port: "3300",
  user: "root",
  password: "tatjana077",
  database: "project1"

});*/

var mysql = require('./mysqlConnection');
const Restaurant = require('./../../models/restaurant/restaurant');

var jsonFromSql = "";// globalna promjenjiva

exports.testPromise = function () { 
  return new Promise(function (resolve, reject) {
    pool.getpool().getConnection(function (err, connection) {
      if (err) reject(err);
      else { 
        connection.query("SELECT * FROM restaurant", function (err, rows) {
         /*  console.log("***" + JSON.stringify(rows)); */
          connection.release();
          resolve(JSON.stringify(rows));
        });
      }
    });npm 
  })
}

exports.getAllRestaurants = function (cb) {
  mysql.getConnection(function(err, connection){
    if(err) throw err;
    else{
      connection.query("SELECT * FROM restaurant WHERE deleted=0", function(err, rows){
        connection.release();
        cb(JSON.stringify(rows));
      })
    }
  })

};

exports.createRestaurant = function(restaurant, cb){
  mysql.getConnection(function(err, connection){
    if(err) cb(err);
    else{
      let sql = "INSERT INTO restaurant (name, address, phone, email, created, deleted) VALUES ?";
      let values = [[restaurant.name, restaurant.address, restaurant.phone, restaurant.email, restaurant.created, 0]];
      connection.query(sql, [values], function(err, rows){
        if(err) cb(err);
        else{
          connection.release();
          restaurant.id=rows.insertId;
          console.log("object object object *************" + restaurant.created);
          cb(restaurant);
        }
      })
    }
  })
};

exports.deleteRestaurant = function(id, cb){
  mysql.getConnection(function(err, connection){
    if(err) cb(err);
    else{
      let sql = "UPDATE restaurant SET deleted=1 WHERE idrestaurant="+id;
      connection.query(sql, function(err, result){
        if(err) cb(err);
        else{
          connection.release();
          cb(result);
        }
      })
    }
  })
}

exports.updateRestaurant = function(restaurant, cb){
  let id = restaurant.id;
  let name = restaurant.name;
  let address = restaurant.address;
  let phone = restaurant.phone;
  let email = restaurant.email;

  mysql.getConnection(function(err, connection){
    if(err) cb(err);
    else{
      let sql = "UPDATE restaurant SET name=?, address=?, phone=?, email=? WHERE idrestaurant=?;";
      let values = [name, address, phone, email, id];
      connection.query(sql, values, function(err, result){
        if(err) cb(err);
        else{
          connection.release();
          cb(result);
        }
      })
    }
  })
}

exports.getRestaurant = function(id, cb) {
  mysql.getConnection(function(err, connection){
    if(err) cb(err);
    else{
      let sql = "SELECT * FROM restaurant WHERE idrestaurant=?;";
      let values = [id];
      connection.query(sql, values, function(err, result) {
        if(err) cb(err);
        else{
          connection.release();
          cb(result);
        }
      })
    }
  })
}



exports.sendSome = function () {
  return "hello";
}
