// datebase/userDB.js
/*var mysql = require('mysql');

var pool = mysql.createPool({
  connectionLimit: 100,
  host: "localhost",
  port: "3300",
  user: "root",
  password: "tatjana077",
  database: "project1"

});*/
const User = require('./../../user/user');
var mysql = require('./mysqlConnection');

exports.addUser = function(user, cb){
    mysql.getConnection(function(err, connection){
        if(err) throw err;
        else{
            let first_name = user.first_name;
            let last_name = user.last_name;
            let username = user.username;
            let phone = user.phone;
            let email = user.email;
            let password = user.password;

            let sqlQuery = "SELECT * FROM user WHERE username = ? LIMIT 1";
            connection.query(sqlQuery, [username], function(err, results){
                if(err) throw err;
                else{
                    
                    if(!results.length){
                        
                        let sql = "INSERT INTO user (first_name, last_name, username, phone, email, password, deleted, role) VALUES ?";
                        let values = [[first_name, last_name, username, phone, email, password, 0, 0]];
                        connection.query(sql, [values], function(err, rows){
                            if(err) cb(err);
                            else{
                                console.log(rows);
                                connection.release();
                             
                                user.id = rows.insertId;
                                cb(user);
                            }
                        })
                    }
                    else{
                        connection.release();
                        let o = {status: "Username exists"};
                        cb(o);
                    }
                }
            })
        }
    })
}
        
  
    /*pool.getConnection(function(err, connection){
        if(err) throw err;
        else{
            var first_name = user.first_name;
            var last_name = user.last_name;
            var sql = "INSERT INTO user (first_name, last_name, username, phone, email, password) VALUES ?";
            var values = [[first_name, last_name, 'tacaa', '065/5222-999', 'blaaa', 'blaaaaa']];
            connection.query(sql, [values], function(err, result){
                if(err) throw err;
                else{
                    connection.release();
                    //console.log(result);
                    cb(JSON.stringify(result));
                }
            
            });
        }
    }); */
    

exports.getAllUsers = function(cb){
   
    mysql.getConnection(function(err, connection){
        if(err) throw err;
        else{
            var sql = "SELECT * FROM user WHERE deleted=0";
            connection.query(sql, function(err, rows){
                if(err) throw err;
                else{
                    connection.release();
                    cb(JSON.stringify(rows));
                }
            })
        }
    })
}

exports.updateUser = function(user, cb){
    let id = user.id;
    let first_name = user.first_name;
    let last_name = user.last_name;
    let username = user.username;
    let phone = user.phone;
    let email = user.email;
    let password = user.password;

    console.log(user.username+"6666666666666666666666666666666666 "+user.id);
    mysql.getConnection(function(err, connection){
        if(err) throw err;
        else{
            var sql = "UPDATE project1.user SET first_name=?, last_name=?, username=?, phone=?, email=?, password=? WHERE iduser=?;";
            let values = [first_name, last_name, username, phone, email, password, id];  
            console.log(values+"77777777777");           
            connection.query(sql, values, function(err, result){
                if(err) cb(err);
                else{
                    console.log(result);
                    connection.release();
                    cb(result);
                }
            })
        }
    })
}

exports.deleteUser = function(id, cb){
    var id = id;
    mysql.getConnection(function(err, connection){
        if(err) throw err;
        else{
            var sql = "UPDATE user SET deleted=1 WHERE iduser="+id;
            connection.query(sql, function(err, result){
                if(err) throw err;
                else{
                    connection.release();
                    cb(JSON.stringify(result));
                }
            });
        }   
    })
}    

exports.loginUser = function(user, cb){
    mysql.getConnection(function(err, connection){
        if(err) throw err;
        else{
            let username = user.username;
            let password = user.password;
            let sql = 'SELECT * FROM user WHERE username = ? AND password = ? AND deleted=0';
            let values = [username, password];
            connection.query(sql, values, function(err, rows){
                if(err) throw err;
                else{
                    connection.release();
                    if(rows[0] !== undefined){
                        
                    var odg = rows[0];
                    console.log('/////// ' + rows[0]);
                    user.iduser = odg.iduser;
                    user.first_name = odg.first_name;
                    user.last_name = odg.last_name;
                    user.phone = odg.phone;
                    user.email = odg.email;
                    user.role = odg.role;
                   // console.log(user);
                   // console.log(odg);
                  //  console.log(rows.length > 0);
                    }
                    cb(user);
                    
                }
            })
        }
    })
}

exports.getUser = function(id, cb){
    mysql.getConnection(function(err, connection){
        if(err) cb(err);
        else{
            let sql = 'SELECT * FROM user WHERE iduser=? AND deleted=0;';
            let values = [id];
            connection.query(sql, values, function(err, result){
                if(err) cb(err);
                else{
                    //console.log(id);
                    connection.release();
                    cb(JSON.stringify(result));
                }
            })
        }
    })
}
